Kernel/Lock.cpp

```diff
@@ -33,11 +33,6 @@ namespace Kernel {
 void Lock::lock()
 {
     ASSERT(!Scheduler::is_active());
-    if (!are_interrupts_enabled()) {
```

Comment:
Why is this being removed? It's still a bug to take a `Lock` while interrupts are disabled.

Reply:
Actually about that, maybe there could be a way to take a lock with interrupts disabled that would fail (assert) if it was already locked? Because in some cases, such as during kernel startup, or with an object I've just created here on the spot, I'm pretty sure noone else has locked the lock, exactly because interrupts are disabled, and i want to do stuff that's normally lock-protected.

Reply:
Sure, we could have separate `Lock` functions for that scenario.

Reply:
Based on irc chats, my understanding was that interrupts where enabled to ensure that the scheduler would still be able to wake up. However, since `lock()` uses `wait_on` which yields, shouldn't this not be an issue?

Also, the requirement that we enable interrupts to get a lock can lead to some iffy code, such as this: https://github.com/SerenityOS/serenity/blob/1dfc66c7cc055d03a87f5851334c5ae2f65d1e2d/Kernel/VM/Region.cpp#L376-L383
which can lead to some pretty crazy race conditions, including the bug mentioned in my above comment.

Reply:
If interrupts are disabled, they are disabled for a reason, and we don't want some code to accidentally re-enable interrupts and pre-empt whatever we're currently doing since that breaks the whole concept of "no interruptions"

Reply:
That code in `handle_zero_fault` is crap indeed and it should be solved somehow differently.